src/main/webapp/app/exam/manage/exercise-groups/exercise-groups.component.ts

```diff
@@ -84,6 +87,34 @@ export class ExerciseGroupsComponent implements OnInit {
         );
     }
 
+    asProgrammingExercise(exercise: Exercise): ProgrammingExercise | null {
+        if (exercise.type === ExerciseType.PROGRAMMING) {
+            return exercise as ProgrammingExercise;
+        }
+        return null;
+    }
+
+    asQuizExercise(exercise: Exercise): QuizExercise | null {
+        if (exercise.type === ExerciseType.QUIZ) {
+            return exercise as QuizExercise;
+        }
+        return null;
+    }
+
+    asTextExercise(exercise: Exercise): TextExercise | null {
+        if (exercise.type === ExerciseType.TEXT) {
+            return exercise as TextExercise;
+        }
+        return null;
+    }
+
+    asModelingExercise(exercise: Exercise): ModelingExercise | null {
+        if (exercise.type === ExerciseType.MODELING) {
+            return exercise as ModelingExercise;
+        }
+        return null;
+    }
+
```

Comment:
I think this is optimal for a switch case. As we have an enum `ExerciseType` defined we could use that to be type-safe for the second parameter of the function:

```suggestion
    caseExercise(exercise: Exercise, exerciseType: ExerciseType): ProgrammingExercise | QuizExercise | TextExercise | ModelingExercise | null {
        switch (exerciseType) {
            case ExerciseType.PROGRAMMING:
                 return exercise as ProgrammingExercise;
            ...
        }
        return null;
    }
    
```

Reply:
The casting methods above are used like:
`<ng-container *ngIf="asSpecificExercise(exercise) as programmingExercise">`
They fulfill two purposes:
1. Cast to a programming exercise if it is one
2. Return null if it isn't a programming exercise

If we have a modeling exercise for example, the switch case statement would do a cast to the modeling exercise but won't return null. ngIf will be true.
Maybe we don't need the casting at all. We won't have typescript tooling support in the template then.

Reply:
https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions
We can stick to just checking just the exercise.type in the template